app/models/space.rb

```diff
@@ -116,6 +116,18 @@ def self.roles
     Space::USER_ROLES.map { |r| Role.find_by_name(r) }
   end
 
+  def last_activitity
+    RecentActivity.where(owner: self).last
+  end
+
+  # Order by when the last activity in the space happened
+  scope :order_by_activity, -> {
```

Comment:
Wouldn't the code below do the same, but with just a single SQL?

```ruby
uniq
  .joins("LEFT JOIN activities ON activities.owner_type = 'Space' AND activities.owner_id = spaces.id")
  .order("activities.created_at DESC")
  ```
Generates something like:

```SQL
Space Load (0.3ms)  SELECT DISTINCT `spaces`.* FROM `spaces` LEFT JOIN activities ON activities.owner_type = 'Space' AND activities.owner_id = spaces.id WHERE `spaces`.`disabled` = 0  ORDER BY activities.created_at DESC
```

If so, `last_activitity` won't be used anymore and can be removed.